### PR TITLE
rsemgr: fix wrong basepath extraction; Fix #171

### LINKED ISSUE #171: rsemanager: Wrong base path extraction
Motivation
----------

>>>from rucio.rse import rsemanager as rsemgr
>>> rse_info = rsemgr.get_rse_info('MOCK')
>>> proto = rsemgr.create_protocol(rse_info, 'write')
>>> proto.parse_pfns(['srm://locahost:8443/srm/managerv2?SFN=/pnfs/data/atlas/atlasgrouptape/group/phys-higgs/mc15_13TeV/group.phys-higgs.mc15_13TeV.mc15c.MGHwpp_tHjb125_yt_minus1.MxAODFlavorSys.p2908.h015.totape_20170825.root/mc15c.MGHwpp_tHjb125_yt_minus1.MxAODFlavorSys.p2908.h015.totape_20170825.root' ])
{'srm://locahost:8443/srm/managerv2?SFN=/pnfs/data/atlas/atlasgrouptape/group/phys-higgs/mc15_13TeV/group.phys-higgs.mc15_13TeV.mc15c.MGHwpp_tHjb125_yt_minus1.MxAODFlavorSys.p2908.h015.totape_20170825.root/mc15c.MGHwpp_tHjb125_yt_minus1.MxAODFlavorSys.p2908.h015.totape_20170825.root': {'name': u'mc15c.MGHwpp_tHjb125_yt_minus1.MxAODFlavorSys.p2908.h015.totape_20170825.root', 'hostname': 'locahos, 'prefix': u'/pnfs/data/atlas/atlasgrouptape/', 'path': u'/group/phys-higgs/mc15_13TeV/group.phys-higgs.mc15_13TeV.', 'scheme': 'srm', 'web_service_path': '/srm/managerv2?SFN=', 'port': '8443'}}

Expected behavior
-----------------
mc15_13TeV.mc15c. vs. mc15_13TeV./mc15c

### PR DESCRIPTION
**Pull request by mlassnig**
rsemgr: fix wrong basepath extraction; Fix #171

### CODE REVIEW

**Comment by vingar** [comment]
test missing :)

**Comment by bari12** [comment]
The parse_pfns method in lib/rucio/rse/protocols/srm.py - shouldn't this be adapted too?

**Comment by vingar** [comment]
if parse_pfns is not defined, the protocol implementation(here srm) will get the one defined in the mother class.

**Comment by bari12** [comment]
Yes, but it is defined in srm.py

File: lib/rucio/tests/test_pfns.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+
+class TestPFNs(object):
+
+    def test_pfn_srm(self):
+        """ PFN (CORE): Test the splitting of PFNs with SRM"""
+
+        rse_info = rsemgr.get_rse_info('MOCK')
+        proto = rsemgr.create_protocol(rse_info, 'read', scheme='srm')
+        pfn = 'srm://mock.com:8443/rucio/tmpdisk/rucio_tests/whatever'
```
Comment (vingar) [inline lib/rucio/tests/test_pfns.py]: `pfns = ['srm://mock.com:8443/rucio/tmpdisk/rucio_tests/whatever',
               'srm://mock.com:8443/srm/managerv2?SFN=/rucio/tmpdisk/rucio_tests/whatever'
               'srm://mock.com:8443/srm/v2/server?SFN=/rucio/tmpdisk/rucio_tests/whatever']
 for pfn in pfns:
    ret = proto...`

**Review by vingar — CHANGES_REQUESTED** [review]

**Comment by vingar** [comment]
maybe it can be removed if the mother one works ?

File: lib/rucio/tests/test_pfns.py (hunk 1 above)
Comment (mlassnig) [reply lib/rucio/tests/test_pfns.py]: well, the asserts are different for all cases

Comment (vingar) [reply lib/rucio/tests/test_pfns.py]: I have maybe overlooked the test but normally the asserts should stay the same

Comment (mlassnig) [reply lib/rucio/tests/test_pfns.py]: they have different ports, hostnames, prefixes, etc across the protocols

Comment (vingar) [reply lib/rucio/tests/test_pfns.py]: ? 
port(8443), hostname(mock.com) and prefixes(/rucio/tmpdisk/rucio_tests/) are the same for the three urls, no ?

Comment (mlassnig) [reply lib/rucio/tests/test_pfns.py]: right, my eyesight is bad

**Comment by vingar** [comment]
please squash commits for the next contribution

**Review by vingar — APPROVED** [review]

**Comment by bari12** [comment]
On a side-note about the commit-squashing. Travis doesn't seem to work well if you squash the commits (or always ammend the same commit). Sometimes it doesn't see that the branch changes and then doesn't re-test the PR... :(

**Pull request merged**